File: net/management-api-v2/cm_api_v2_clone_environment.cs
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;
using Kontent.Ai.Management.Models.Environments;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "KONTENT_AI_MANAGEMENT_API_KEY",
    EnvironmentId = "KONTENT_AI_ENVIRONMENT_ID"
});

var response = await client.CloneEnvironmentAsync(new EnvironmentCloneModel
{
    Name = "New environment",
    RolesToActivate = new[]
    {
        Guid.Parse("ee483b59-5a24-4010-b277-ae224c34bc71")
    },
    CopyDataOptions = new CopyDataOptions {
        ContentItemsAssets = true,
        ContentItemVersionHistory = false
    }
});

File: js/management-api-v2/cm_api_v2_clone_environment.js
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
// Using ES6 syntax
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
    environmentId: '<YOUR_ENVIRONMENT_ID>',
    apiKey: '<YOUR_API_KEY>'
});

const response = await client.cloneEnvironment()
    .withData(
        {
            name: "Develop",
            roles_to_activate: ["ee483b59-5a24-4010-b277-ae224c34bc71"]
        }
    )
    .toPromise();
